### PR TITLE
Bugfix - permissible purposes

### PR DESCRIPTION
**Pull request by joefitter**
* Added permissible purposes to PDF and granted view for new PPLs
* Updated legacy to use permissible purposes field type to show answers in context

### CODE REVIEW

File: views/pdf/v1/index.jsx [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-    <StandardConditions />
-  </Fragment>
-)
-
-const mapStateToProps = ({ project }) => ({ project });
-
-export default connect(mapStateToProps)(Modern);
+export default function PDF() {
+  const project = useSelector(state => state.project);
+  const sections = Object.values(getGrantedSubsections(1)).filter(s => !s.granted.show || s.granted.show(project)).sort((a, b) => a.granted.order - b.granted.order);
```
Comment (lennym) [inline views/pdf/v1/index.jsx]:
```suggestion
  const sections = Object.values(getGrantedSubsections(1))
    .filter(s => !s.granted.show || s.granted.show(project))
    .sort((a, b) => a.granted.order - b.granted.order);
```

Comment (lennym) [reply views/pdf/v1/index.jsx]: `s => !s.granted.show` looks like it's going to include if `show: false` which feels like the opposite of what we want.

Perhaps `show === undefined`?

File: client/schema/v0/index.js [hunk 2]
```diff
@@ -203,8 +203,9 @@ export default () => ({
           {
             name: 'purpose',
             label: 'Which purposes apply to your project?',
-            type: 'checkbox',
+            type: 'permissible-purpose',
             className: 'smaller',
+            preserveHeirarchy: true,
```
Comment (lennym) [inline client/schema/v0/index.js]: How did this ever get through?!

`preserveHierarchy` damnit!

File: views/pdf/v1/index.jsx (hunk 1 above)
Comment (joefitter) [reply views/pdf/v1/index.jsx]: not sure, I didn't change this

File: client/schema/v0/index.js (hunk 2 above)
Comment (joefitter) [reply client/schema/v0/index.js]: HEHE

File: views/pdf/v1/index.jsx (hunk 1 above)
Comment (joefitter) [reply views/pdf/v1/index.jsx]: I _think_ show is _always_ a function or undefined

**Pull request merged**